xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/TestUtils.java

```diff
@@ -1427,6 +1432,46 @@ public String getSecretToken()
         return this.secretToken;
     }
 
+    /**
+     * Get the WCAG context for the test suite.
+     *
+     * @return WCAG Context
+     */
+    public WCAGContext getWcagContext()
+    {
+        return this.wcagContext;
+    }
+
+    public void writeWCAGResults(Logger logger) throws IOException{
+        WCAGContext wcagContext = getWcagContext();
+        if (wcagContext.isWcagEnabled()) {
+            float totalTime = (float) wcagContext.wcagTimer / 1000;
+            logger.info("Time spent on WCAG validation [{}] (in s)", totalTime);
+            File wcagDir = new File("target/wcag");
+            if (wcagContext.hasWCAGWarnings()) {
+                logger.warn("There are some accessibility issues in the test suite. " +
+                  "See target/wcag/wcagWarnings for more details.");
+                if (!wcagDir.exists()) {
+                    Files.createDirectory(wcagDir.toPath());
+                }
+                String outputName = "wcagWarnings";
+                File warningsFile = new File(wcagDir, outputName);
+                WCAGContext.writeWCAGReportToFile(warningsFile, wcagContext.buildWarningsReport());
+            }
+            if (wcagContext.hasWCAGFails()) {
+                if (!wcagDir.exists()) {
+                    Files.createDirectory(wcagDir.toPath());
+                }
+                String outputName = "wcagFails";
+                File failsFile = new File(wcagDir, outputName);
+                WCAGContext.writeWCAGReportToFile(failsFile, wcagContext.buildFailsReport());
+            }
+            // Assert the results of the different WCAG Checks are all empty
+            assertFalse(wcagContext.hasWCAGFails(), "There are some accessibility issues in the test suite." +
+              "See target/wcag/wcagFails for more details.");
```

Comment:
To me it seems unexpected that the code to write the results contains the assertion that fails the build. Also, wouldn't it be nicer to include at least the description of the first failure in the assertion such that it is displayed in Jenkins and whoever sees this doesn't always need to look at the output files?